homeassistant/components/fritz/common.py

```diff
@@ -260,10 +252,92 @@ class FritzData:
     """Storage class for platform global data."""
 
     tracked: dict = field(default_factory=dict)
+    profile_switches: dict = field(default_factory=dict)
+
+
+class FritzDeviceBase(Entity):
+    """Entity base class for a device connected to a FRITZ!Box router."""
+
+    def __init__(self, router: FritzBoxTools, device: FritzDevice) -> None:
+        """Initialize a FRITZ!Box device."""
+        self._router = router
+        self._mac: str = device.mac_address
+        self._name: str = device.hostname or DEFAULT_DEVICE_NAME
+
+    @property
+    def name(self) -> str:
+        """Return device name."""
+        return self._name
+
+    @property
+    def ip_address(self) -> str | None:
+        """Return the primary ip address of the device."""
+        if self._mac:
+            device: FritzDevice = self._router.devices[self._mac]
+            return device.ip_address
+        return None
+
+    @property
+    def mac_address(self) -> str:
+        """Return the mac address of the device."""
+        return self._mac
+
+    @property
+    def hostname(self) -> str | None:
+        """Return hostname of the device."""
+        if self._mac:
+            device: FritzDevice = self._router.devices[self._mac]
+            return device.hostname
+        return None
+
+    @property
+    def device_info(self) -> DeviceInfo:
+        """Return the device information."""
+        return {
+            "connections": {(CONNECTION_NETWORK_MAC, self._mac)},
+            "identifiers": {(DOMAIN, self._mac)},
+            "default_name": self.name,
+            "default_manufacturer": "AVM",
+            "default_model": "FRITZ!Box Tracked device",
+            "via_device": (
+                DOMAIN,
+                self._router.unique_id,
+            ),
+        }
+
+    @property
+    def should_poll(self) -> bool:
+        """No polling needed."""
+        return False
+
+    @property
+    def entity_registry_enabled_default(self) -> bool:
+        """Return if the entity should be enabled when first added to the entity registry."""
+        return False
+
+    async def async_process_update(self) -> None:
+        """Update device."""
+        raise NotImplementedError()
+
+    async def async_on_demand_update(self) -> None:
+        """Update state."""
+        await self.async_process_update()
+        await self.async_update_ha_state()
```

Comment:
```suggestion
        self.async_write_ha_state()
```

Any reason to not write the state directly?